pilot/cmd/pilot-agent/main.go

```diff
@@ -57,6 +57,9 @@ const (
 	trustworthyJWTPath = "./var/run/secrets/tokens/istio-token"
 	localHostIPv4      = "127.0.0.1"
 	localHostIPv6      = "[::1]"
+
+	// Similar with ISTIO_META_, which is used to customize the node metadata - this customizes extra header.
```

Comment:
can you add some more comments about the format of the header value and how it is being used.

Reply:
Nothing special - in envoy you can add some headers, we lost this ability here. Each server may have its own extra headers ( apikeys, custom params, etc ). The format is similar with ISTIO_META_ - prefix is removed, the rest is header name and value.

Reply:
I think it just provides an alternative way to pass metadata. Besides for auth, is there any use case?

Reply:
@hzxuzhonghu we have also looked into a gateway that does  header based routing to different Istiod instances. but primary case is auth